From 64fd25af5fea3a3c9c12122a2fd0fab3cd420f72 Mon Sep 17 00:00:00 2001
From: asmitk01-in <asmitk01@in.ibm.com>
Date: Wed, 5 Feb 2020 00:14:42 -0600
Subject: [PATCH] libmimetic: Removing test directory from the Makefile

The test directory is removed as there were errors (relocation errors)
since the shared object files of libc++ and libc were not taken from the
sysroot path but from the system's library path where the required
version of the .so is not present.

The files 'mimetic/codec/other_codecs.h' and 'mimetic/contenttype.cxx'
contains changes that resolves errors when compiling on target system
aarch64-yoe-linux-musl.

Signed-off-by: asmitk01-in <asmitk01@in.ibm.com>
Change-Id: I1822b36ad4a1fa5e7b165d8cac3a6e96aca11c35
---
 Makefile.am                  | 2 +-
 mimetic/codec/other_codecs.h | 4 ++++
 mimetic/contenttype.cxx      | 4 ++++
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 634bd4c..c7cbae7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 MAINTAINERCLEANFILES = Makefile Makefile.in config.cache
 EXTRA_DIST=LICENSE m4
-SUBDIRS = mimetic doc examples test win32
+SUBDIRS = mimetic doc examples win32
 INCLUDES=-I$(top_srcdir)
 ACLOCAL_AMFLAGS=-I m4

diff --git a/mimetic/codec/other_codecs.h b/mimetic/codec/other_codecs.h
index c4ccf3f..0fc88fd 100644
--- a/mimetic/codec/other_codecs.h
+++ b/mimetic/codec/other_codecs.h
@@ -8,6 +8,10 @@
 #define _MIMETIC_CODEC_OTHER_CODECS_H_
 #include <mimetic/codec/codec_base.h>

+#ifndef __GLIBC__
+    typedef unsigned int uint;
+#endif
+
 namespace mimetic
 {

diff --git a/mimetic/contenttype.cxx b/mimetic/contenttype.cxx
index 522082d..0577d44 100644
--- a/mimetic/contenttype.cxx
+++ b/mimetic/contenttype.cxx
@@ -13,6 +13,10 @@
 #include <mimetic/tokenizer.h>
 #include <mimetic/utils.h>

+#ifndef __GLIBC__
+    typedef unsigned int uint;
+#endif
+
 namespace mimetic
 {
 using namespace std;
-- 
2.9.3

